src/context-items/custom-validation.ts

```diff
@@ -19,6 +19,10 @@ export class CustomValidation implements ContextItem {
       if ((!isPromise && failed) || (isPromise && this.negated)) {
         context.addError(this.message, value, meta);
       }
+      // rename field if return type is string
+      if (typeof actualResult === 'string') {
+        context.renameFieldInstance(actualResult, meta);
```

Comment:
May I ask why you thought of making use of the custom validators?

Changing the meaning of the return of a custom validator will be quite disruptive for everybody, and is certainly a breaking change.

Perhaps a new function in the chain would be a better idea?